### PR TITLE
Updated mission_data_downloader.ipynb and  mission_data_downloader.py

### PR DESCRIPTION
**Pull request by luigigisolfi**
This pull request supersedes PR #82 

Implemented Changes wrt previous PR:
1) added Grail-A and Grail-B among the supported missions
2) show non-supported mission functionalities
3) show meta-kernel reader functionalities
4) showcase cassini flyby_IDs choice
5) fixed markdowns based on Dominic suggestions in #82

### CODE REVIEW

File: estimation/mission_data_downloader.py [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+# ## Objectives
+# 
+# - This script shows how to use the TudatPy `mission_data_downloader` functionality  to download:
+# 1) **SPICE** (clock, frame, orientation, planetary) **Kernels**
+# 2) **Closed Loop Doppler Data** (DSN and IFMS)
+# 3) **Ancillary** Ionospheric and Tropospheric **Data**
+# 
+# for some supported missions (see **NOTE 1** for a list of supported missions). 
+# 
+# - Moreover, we also show how to use the functionality to download:
```
Comment (DominicDirkx) [inline estimation/mission_data_downloader.py:13]: Here, you give the exact same list as lines 7-9, seems superfluous

File: estimation/mission_data_downloader.py [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+# 
+# - Moreover, we also show how to use the functionality to download: 
+# 1) **SPICE** (clock, frame, orientation, planetary) **Kernels**
+# 2) **Closed Loop Doppler Data** (DSN and IFMS)
+# 3) **Ancillary** Ionospheric and Tropospheric **Data**
+# 
+# for a custom, user-defined mission.
+# 
+# - Last but not least, we show how to use tudatpy to download:
+#   **all files listed in a given Meta-Kernel**.
```
Comment (DominicDirkx) [inline estimation/mission_data_downloader.py:21]: mention that it's a SPICE kernel

File: estimation/mission_data_downloader.py [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+# **NOTE 2. (Default and Custom Outputs)**
+# 
+# The downloaded data for a given mission is stored (by default) in a folder named `<mission_name>_archive`. Subfolders are created automatically where Kernels, Radio Science Data and Ancillary Data are stored. 
+# The user can still define a custom output path passing the flag `custom_output` in the `get_mission_files` function. The output files will then be found in: `custom_path`
+# 
+# **NOTE 3. (About DSN-TNF)**
+# 
+# No DSN-TNF download functionality has been implemented (at least, not yet!)
+# 
+# **Note 4. About spiceypy**
```
Comment (DominicDirkx) [inline estimation/mission_data_downloader.py:56]: I think you can leave this out. Once merged, we'll have spiceypy as a dependency, and any new user will have it automatically. This line makes it seem as though you need a 'special' tudaty version

File: estimation/mission_data_downloader.py [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+#print(f'JUICE Radio Science: {radio_science_files_juice}\n') # it will be empty for now... (no Radio Science yet available on the server)
+#print(f'JUICE Ancillary: {ancillary_files_juice}\n') # it will be empty for now... (no Ancillary files yet available on the server)
+
+
+# ## Cassini Downloader (with Custom Output)
+# 
+# ### Set Flybys for Downloading
+# 
+# The most valuable data collected for Cassini is probably the one related to the various flybys of the Moons of Saturn.
+# For this reason, in order to retrieve Cassini data, we will require the **name of the flyby** data the user wishes to download, rather than a start and end date (see **Notes 6 and 7** for info on the supported flybys).
```
Comment (DominicDirkx) [inline estimation/mission_data_downloader.py:144]: Make a note that (for now!) we only have the Titan flybys implemented

File: estimation/mission_data_downloader.py (hunk 2 above)
Comment (luigigisolfi) [reply estimation/mission_data_downloader.py:21]: changed to
- Last but not least, we show how to use tudatpy to download:
  **all files listed in a given SPICE Meta-Kernel**.

File: estimation/mission_data_downloader.py [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+# 
+# 1) **existing spice kernels** in the mission folder
+# 2) (new) **downloaded spice kernels**
+# 3) **standard TUDAT kernels** from [`spice.load_standard_kernels()`](https://py.api.tudat.space/en/latest/spice.html#tudatpy.interface.spice.load_standard_kernels) 
+# 
+# List of currently supported input_missions:
+# - **Mars Reconnnaissance Orbiter (MRO)** [Available: Spice Kernels, Doppler Data, Ancillary]
+# - **Mars Express (MEX)** [Available: Spice Kernels, Doppler Data, Ancillary]
+# - **Jupiter Icy Moons Explorer (JUICE)** [Available: Spice Kernels, Not Available (yet): Doppler Data, Ancillary]
+# - **Cassini**
```
[the remark this answers is not in the document]
Comment (luigigisolfi) [reply estimation/mission_data_downloader.py:38]: you meant Cassini and Grail, right? 

List of currently supported input_missions:
- **Mars Reconnnaissance Orbiter (MRO)** [Available: Spice Kernels, Doppler Data, Ancillary]
- **Mars Express (MEX)** [Available: Spice Kernels, Doppler Data, Ancillary]
- **Jupiter Icy Moons Explorer (JUICE)** [Available: Spice Kernels, Not Available (yet): Doppler Data, Ancillary]
- **Cassini** [Available: Spice Kernels, Doppler Data, Ancillary for all **Titan Flybys**]
- **GRAIL (both Grail-A and Grail-B)** [Available: Spice Kernels, Doppler Data, Ancillary]

**Comment by DominicDirkx** [comment]
Looks great, and runs like a charm! I just needed to install two additional dependencies for these lines:

       colored_headers = [f"{Fore.MAGENTA}{header}{Fore.RESET}" for header in headers]
        print(tabulate(data, colored_headers, tablefmt="fancy_grid", stralign="center"))

conda install tabulate
conda install colorama

I'm not familiar with these dependencies, but I'm fine adding them. Just to be clear: these are to make the tabular user output more visually appealing right?

I think I'll also add part of this example into a python test, to make sure that everything that is set up now continues to work well in the future.

**Comment by luigigisolfi** [comment]
_> I'm not familiar with these dependencies, but I'm fine adding them. Just to be clear: these are to make the tabular user output more visually appealing right?_

Yes, correct!

> I think I'll also add part of this example into a python test, to make sure that everything that is set up now continues to work well in the future.

Yeah, that would be nice as well.

**Pull request merged**